src/screens/MainApp/Home/Dashboard/DiagnosesBoard/index.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState, useEffect } from 'react'
+import { View, Text, SafeAreaView, ActivityIndicator } from 'react-native'
+import styles from './styles'
+import BoardItem from './BoardItem'
+import { getDiagnosesFromCurrentUser } from '~/databaseService'
+import * as StorageService from '~/storageService'
+import { ForceRerenderOnNavigation } from '~/navigationService'
+import { ScrollView } from 'react-native-gesture-handler'
+
+const getURL = async (imageReference) => {
+  try {
+    return await StorageService.getDownloadURL(imageReference)
+  } catch (error) {
+    return null
+  }
+}
+
+const buildBoardItems = async (downloadedData) => (
+  downloadedData.reduce(
+    async (boardItems, diagnose, index) => {
+      boardItems = await boardItems
+      const boardItem = await getBoardItemFromDiagnose(diagnose, index)
+      boardItems.push(boardItem)
+
+      return boardItems
+    },
+    Promise.resolve([])
+  )
+)
+
+const getBoardItemFromDiagnose = async (diagnose, key) => {
+  const thumbnail = await getURL(diagnose.imageReferences[0])
+  return (
+    <BoardItem
+      key={key}
+      thumbnail={thumbnail}
+      diagnose={diagnose}
+    />
+  )
+}
+
+export const sortByCreatedAt = (diagnoses) => (
```

Comment:
Careful when using previous code. This `export` is from a utility module.

Reply:
It would be nice to have that function in a different place since we both use it.